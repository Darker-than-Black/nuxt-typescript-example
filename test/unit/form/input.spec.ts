import VueTestInstance from '~/test/factory/vue-instance'
import Input from '~/components/form/input.vue'

describe('input.vue', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = VueTestInstance(Input)
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check input function', () => {
    const event: any = { target: { value: 'test text' } }
    expect(wrapper.vm.input(event)).toEqual(event.target.value)
  })
})
