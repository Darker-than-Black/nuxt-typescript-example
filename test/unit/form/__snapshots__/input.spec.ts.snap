// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input.vue should match the snapshot 1`] = `
<field-wrapper-stub label="">
  <div class="position-relative"><input id="" name="" placeholder="" type="text" list="" class="input"> </div>
</field-wrapper-stub>
`;
